builtin/logical/aws/path_config_root.go

```diff
@@ -204,6 +214,41 @@ func (b *backend) pathConfigRootWrite(ctx context.Context, req *logical.Request,
 		return nil, err
 	}
 
+	// Now that the root config is set up, register the rotation job if it required
+	if rc.RotationSchedule != "" || rc.RotationTTL != 0 {
+		cfgReq := &logical.RotationJobConfigureRequest{
+			Name:             rootRotationJobName,
+			MountPoint:       req.MountPoint,
+			ReqPath:          req.Path,
+			RotationSchedule: rc.RotationSchedule,
+			RotationWindow:   rc.RotationWindow,
+			RotationTTL:      rc.RotationTTL,
+		}
+
+		rotationJob, err := logical.ConfigureRotationJob(cfgReq)
+		if err != nil {
+			return logical.ErrorResponse("error configuring rotation job: %s", err), nil
+		}
+
+		b.Logger().Debug("Registering rotation job", "mount", req.MountPoint+req.Path)
+		rotationID, err := b.System().RegisterRotationJob(ctx, rotationJob)
```

Comment:
How and when does a rotation job get deregistered?

Reply:
Good call out, not sure if we have yet discussed the idea of deregistering rotation. Is it something we should allow? I wonder if there's any differences between allowing automated rotation to be disabled versus a pause mechanism meant only for emergencies?

Reply:
De-registration / off-boarding should be allowed IMO, if only as a disabling escape hatch. Added a method on the SystemView to Deregister a rotation job. This will be triggered if a boolean `disable_automated_rotation` is set to true on the config. Example of this in AWS is linked [here](https://github.com/hashicorp/vault/pull/29273/files#diff-11906daf04096f16c364abff4761df97c3b879915152cee7f14a03e1f74dc32eR242-R253)

Reply:
I was thinking about the disable and delete actions. Without deregistration, how do we stop rotations when a mount is disabled or a static role is deleted that have been registered with the Rotation Manager?

Reply:
Ah I see, that's a good question! So this won't be visible in this PR, but the Enterprise Rotation Manager implementation does store the request path that a credential comes from. The RM also has access to `core.Router`, and I believe we should be able to use `MatchingMountEntry` to determine if a given mount exists.

This will be performed on the Enterprise Rotation Manager code (before it routes the request to the plugin backend, it can confirm that a matching mount entry exists for handling the request). I can make sure to test that flow, and call it out in the Enterprise PR once it is opened (after this PR is merged in so that CI in Enterprise is happy)

Reply:
Yup! We can deregister the job if we find no Matching Mount Entry for that path, and will log a warning to the user letting them know that the backend/role is no longer active